lib/rules/no-whitespace-before-property.js

```diff
@@ -25,14 +61,20 @@ module.exports = function(context) {
             var prop = node.property;
 
             if (astUtils.isTokenOnSameLine(obj, prop)) {
-                if (sourceCode.isSpaceBetweenTokens(obj, prop)) {
-                    context.report({
-                        node: node,
-                        message: "Unexpected whitespace before property '{{ propName }}'.",
-                        data: {
-                            propName: prop.name
-                        }
-                    });
+                var punc = sourceCode.getTokenBefore(prop);
+
+                if (punc.type !== "Punctuator") {
+                    return;
```

Comment:
It shouldn't, no - was trying to be defensive in case something outside the rule goes wrong. Do you feel like that's unnecessary?